AirCasting/Graph/GraphView.swift

```diff
@@ -36,35 +36,81 @@ struct GraphView<StatsViewModelType>: View where StatsViewModelType: StatisticsC
-                              }
-                        StatisticsContainerView(statsContainerViewModel: statsContainerViewModel,
-                                                threshold: threshold)
-                    }
+           
+            if isProceeding(session: session) {
+                    if let threshold = thresholds.threshold(for: selectedStream) {
+                        ZStack(alignment: .topLeading) {
+                            if let selectedStream = selectedStream {
+                                Graph(stream: selectedStream,
+                                      thresholds: threshold,
+                                      isAutozoomEnabled: session.type == .mobile).onDateRangeChange { [weak graphStatsDataSource, weak statsContainerViewModel] range in
+                                        graphStatsDataSource?.dateRange = range
+                                        statsContainerViewModel?.adjustForNewData()
+                                      }
+                                StatisticsContainerView(statsContainerViewModel: statsContainerViewModel,
+                                                        threshold: threshold)
+                            }
 
-                }
-                NavigationLink(destination: HeatmapSettingsView(changedThresholdValues: threshold.rawThresholdsBinding)) {
-                    EditButtonView()
-                }
-                .padding(.horizontal)
-                .padding(.top)
-                
-                ThresholdsSliderView(threshold: threshold)
-                    .padding()
-                    // Fixes labels covered by tabbar
-                    .padding(.bottom)
+                        }
+                        HStack() {
+                            startTimeText
+                            Spacer()
+                            endTimeText
+                        }
+                        .foregroundColor(.aircastingGray)
+                        .padding(.horizontal, 5)
+                        NavigationLink(destination: HeatmapSettingsView(changedThresholdValues: threshold.rawThresholdsBinding)) {
+                            EditButtonView()
+                        }
+                        .padding(.horizontal)
+                        .padding(.top)
+                        
+                        ThresholdsSliderView(threshold: threshold)
+                            .padding()
+                            // Fixes labels covered by tabbar
+                            .padding(.bottom)
+                    }
             }
             Spacer()
         }
         .navigationBarTitleDisplayMode(.inline)
     }
+    
+    var startTimeText: some View {
+        // Active mobile sessions are being represented in device time, so no UTC reformatting is needed
+        // Other sessions are already in the UTC format
+        var formatter = DateFormatters.GraphView.dataFormatter
```

Comment:
reassigning in conditionals is considered bad practice and rarely needed. Now that we simplified date formatter we can do it in a single assignment:
```Swift
let isActiveMobile = session.isMobile && sesssion.isActive
let formatter = isActiveMobile ? DateFormatters.GraphView.dataFormatter : DateFormatters.GraphView.mobileActiveFormatter
```

(same in `endTimeText`)